addons/payment_sips/models/sips.py

```diff
@@ -0,0 +1,243 @@
+# -*- coding: utf-'8' "-*-"
+
+try:
+    import simplejson as json
+except ImportError:
+    import json
+import logging
+from hashlib import sha256
+import urlparse
+import unicodedata
+
+from openerp.addons.payment.models.payment_acquirer import ValidationError
+from openerp.addons.payment_sips.controllers.main import SipsController
+from openerp.osv import osv, fields
+from openerp.tools.float_utils import float_compare
+from openerp import SUPERUSER_ID
+
+_logger = logging.getLogger(__name__)
+
+
+UNORMALIZE_CHARS = {
+    u'Š': u'S', u'š': u's', u'Đ': u'Dj', u'đ': u'dj', u'Ž': u'Z', u'ž': u'z',
+    u'Č': u'C', u'č': u'c', u'Ć': u'C', u'ć': u'c', u'À': u'A', u'Á': u'A', u'Â': u'A', u'Ã': u'A',
+    u'Ä': u'A', u'Å': u'A', u'Æ': u'A', u'Ç': u'C', u'È': u'E', u'É': u'E', u'Ê': u'E', u'Ë': u'E',
+    u'Ì': u'I', u'Í': u'I', u'Î': u'I', u'Ï': u'I', u'Ñ': u'N', u'Ò': u'O', u'Ó': u'O', u'Ô': u'O',
+    u'Õ': u'O', u'Ö': u'O', u'Ø': u'O', u'Ù': u'U', u'Ú': u'U', u'Û': u'U', u'Ü': u'U', u'Ý': u'Y',
+    u'Þ': u'B', u'ß': u'Ss', u'à': u'a', u'á': u'a', u'â': u'a', u'ã': u'a', u'ä': u'a', u'å': u'a',
+    u'æ': u'a', u'ç': u'c', u'è': u'e', u'é': u'e', u'ê': u'e', u'ë': u'e', u'ì': u'i', u'í': u'i',
+    u'î': u'i', u'ï': u'i', u'ð': u'o', u'ñ': u'n', u'ò': u'o', u'ó': u'o', u'ô': u'o', u'õ': u'o',
+    u'ö': u'o', u'ø': u'o', u'ù': u'u', u'ú': u'u', u'û': u'u', u'ý': u'y', u'ý': u'y', u'þ': u'b',
+    u'ÿ': u'y', u'Ŕ': u'R', u'ŕ': u'r', u"`": u"'", u"´": u"'", u"„": u",", u"`": u"'",
+    u"´": u"'", u"“": u"\"", u"”": u"\"", u"´": u"'", u"&acirc;€™": u"'", u"{": u"",
+    u"~": u"", u"–": u"-", u"’": u"'", u">": u" ", u"<": u" "
+}
+
+
+def unormalize(text):
+    if not text:
+        return text
+
+    for inv_char, valid_char in UNORMALIZE_CHARS.items():
```

Comment:
This module doesn't depend on mail, by replace you mean calling it from mail or copy/paste it?